base/gcd.h

```diff
@@ -0,0 +1,40 @@
+// LAF Base Library
+// Copyright (c) 2019 Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifndef BASE_GCD_H_INCLUDED
+#define BASE_GCD_H_INCLUDED
+#pragma once
+
+namespace base {
+
+  template<typename T>
+  T gcd(T a, T b) {
+    T output = 1;
```

Comment:
Instead of using an `output` variable we could just `return value;` on each case.